### PR TITLE
[MTDSA 12351] Models for  Retrieve a Historic FHL UK Property Income & Expenses Period Summary

### PR DESCRIPTION
**Pull request by sean-flynn-HMRC**

### CODE REVIEW

**Review by SamuelJacobsHMRC — CHANGES_REQUESTED** [review]

File: test/v2/models/response/retrieveHistoricFhlUkPiePeriodSummary/RentARoomSpec.scala [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    """{
+      |      "amountClaimed":5000.99
+      |    }
+      |""".stripMargin
+  )
+
+
+  "reads" when {
+    "passed a valid Income JSON" should {
+      "return a valid model" in {
```
Comment (mattstephens-hmrc) [inline test/v2/models/response/retrieveHistoricFhlUkPiePeriodSummary/RentARoomSpec.scala:61]: Please can these be "return a valid object" and "passed a valid Expenses object"? "Model" refers to the group of classes (definitions)

File: app/v2/models/response/retrieveHistoricFhlUkPiePeriodSummary/RetrieveHistoricFhlUkPiePeriodSummaryResponse.scala [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+                                                         expenses: Option[PeriodExpenses])
+
+object RetrieveHistoricFhlUkPiePeriodSummaryResponse extends HateoasLinks {
+  implicit val writes: OWrites[RetrieveHistoricFhlUkPiePeriodSummaryResponse] = Json.writes[RetrieveHistoricFhlUkPiePeriodSummaryResponse]
+
+  implicit val reads: Reads[RetrieveHistoricFhlUkPiePeriodSummaryResponse] = (
+    (JsPath \ "from").read[String] and
+      (JsPath \ "to").read[String] and
+      (__ \ "financials" \ "incomes" ).readNullable[PeriodIncome] and
+      //(JsPath \ "financials/incomes").readNullable[PeriodIncome] and
```
[the remark this answers is not in the document]
Comment (sean-flynn-HMRC) [reply app/v2/models/response/retrieveHistoricFhlUkPiePeriodSummary/RetrieveHistoricFhlUkPiePeriodSummaryResponse.scala:37]: Done

File: app/v2/models/request/retrieveHistoricFhlUkPiePeriodSummary/RetrieveHistoricFhlUkPiePeriodSummaryRequest.scala [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+ */
+
+package v2.models.request.retrieveHistoricFhlUkPropertyAnnualSubmission
+
+import v2.models.domain.{Nino}
+import v2.models.request.RawData
+
+case class RetrieveHistoricFhlUkPiePeriodSummaryRawData(nino: String, periodId: String) extends RawData
+
+case class RetrieveHistoricFhlUkPiePeriodSummaryRequest(nino: Nino, periodId: String)
```
[the remark this answers is not in the document]
Comment (sean-flynn-HMRC) [reply app/v2/models/request/retrieveHistoricFhlUkPiePeriodSummary/RetrieveHistoricFhlUkPiePeriodSummaryRequest.scala:24]: Added

File: test/v2/models/response/retrieveHistoricFhlUkPiePeriodSummary/RetrieveHistoricFhlUkPiePeriodSummaryResponseSpec.scala [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+      |         "residentialFinancialCostsCarriedForward": 5000.99,
+      |         "ukRentARoom": {
+      |            "amountClaimed": 5000.99
+      |         }
+      |      }
+      |   }
+      |}
+      |""".stripMargin)
+
+//  private def decimal(value: String): Option[BigDecimal] = Option(BigDecimal(value))
```
[the remark this answers is not in the document]
Comment (sean-flynn-HMRC) [reply test/v2/models/response/retrieveHistoricFhlUkPiePeriodSummary/RetrieveHistoricFhlUkPiePeriodSummaryResponseSpec.scala:60]: Done

File: test/v2/models/response/retrieveHistoricFhlUkPiePeriodSummary/RetrieveHistoricFhlUkPiePeriodSummaryResponseSpec.scala [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+        Some(5000.99),
+        Some(5000.99),
+        Some(5000.99),
+        Some(5000.99),
+        Some(5000.99),
+        Some(RentARoomExpenses(Some(5000.99)))
+      ))
+  )
+
+//  val model = RetrieveHistoricFhlUkPiePeriodSummaryResponse(
```
[the remark this answers is not in the document]
Comment (sean-flynn-HMRC) [reply test/v2/models/response/retrieveHistoricFhlUkPiePeriodSummary/RetrieveHistoricFhlUkPiePeriodSummaryResponseSpec.scala:82]: Done

File: test/v2/models/response/retrieveHistoricFhlUkPiePeriodSummary/RetrieveHistoricFhlUkPiePeriodSummaryResponseSpec.scala [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+      |}
+      |""".stripMargin)
+
+//  private def decimal(value: String): Option[BigDecimal] = Option(BigDecimal(value))
+
+  val model = RetrieveHistoricFhlUkPiePeriodSummaryResponse(
+    fromDate = "2001-01-01",
+    toDate = "2001-01-01",
+    income = Some(
+      PeriodIncome(Some(5000.99), Some(5000.99), Option(RentARoomIncome(Some(5000.99))))
```
[the remark this answers is not in the document]
Comment (sean-flynn-HMRC) [reply test/v2/models/response/retrieveHistoricFhlUkPiePeriodSummary/RetrieveHistoricFhlUkPiePeriodSummaryResponseSpec.scala:66]: Good idea

File: test/v2/models/response/retrieveHistoricFhlUkPiePeriodSummary/RetrieveHistoricFhlUkPiePeriodSummaryResponseSpec.scala [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+    ),
+    expenses = Some(
+      PeriodExpenses(
```
[the remark this answers is not in the document]
Comment (sean-flynn-HMRC) [reply test/v2/models/response/retrieveHistoricFhlUkPiePeriodSummary/RetrieveHistoricFhlUkPiePeriodSummaryResponseSpec.scala:69]: Done

File: test/v2/models/response/retrieveHistoricFhlUkPiePeriodSummary/RentARoomSpec.scala (hunk 1 above)
Comment (sean-flynn-HMRC) [reply test/v2/models/response/retrieveHistoricFhlUkPiePeriodSummary/RentARoomSpec.scala]: Done

**Pull request merged**